File: 2014/AdvancedTrack/_03-Hands On - Handling Solid Edge Events/cs/Example1/Example1/Form1.cs
using SolidEdgeContrib;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace Example1
{
    public partial class Form1 : Form
    {
        private SolidEdgeFramework.Application _application = null;
        private ApplicationEventWatcher _applicationEventWatcher = null;
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            OleMessageFilter.Register();
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e)
        {
            if (_applicationEventWatcher != null)
            {
                _applicationEventWatcher.Dispose();
                _applicationEventWatcher = null;
            }
            _application = null;
        }

        private void exitToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void eventButton_Click(object sender, EventArgs e)
        {
            try
            {
                if (eventButton.Checked)
                {
                    if (_application == null)
                    {
                        // On a system where Solid Edge is installed, the COM ProgID will be
                        // defined in registry: HKEY_CLASSES_ROOT\SolidEdge.Application
                        Type t = Type.GetTypeFromProgID(SolidEdge.PROGID.Application, throwOnError: true);

                        // Using the discovered Type, create and return a new instance of Solid Edge.
                        _application = (SolidEdgeFramework.Application)Activator.CreateInstance(type: t);

                        // Show Solid Edge.
                        _application.Visible = true;
                    }

                    _applicationEventWatcher = new ApplicationEventWatcher(this, _application);
                }
                else
                {
                    _applicationEventWatcher.Dispose();
                    _applicationEventWatcher = null;
                    _application = null;
                }
            }
            catch (System.Exception ex)
            {
                MessageBox.Show(this, ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
        }

        private void clearButton_Click(object sender, EventArgs e)
        {
            eventLogTextBox.Clear();
        }

        #region SolidEdgeFramework.ISEApplicationEvents

        public void OnAfterActiveDocumentChange(object theDocument)
        {
            eventLogTextBox.AppendText("OnAfterActiveDocumentChange");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnAfterCommandRun(int theCommandID)
        {
            eventLogTextBox.AppendText("OnAfterCommandRun");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnAfterDocumentOpen(object theDocument)
        {
            eventLogTextBox.AppendText("OnAfterDocumentOpen");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnAfterDocumentPrint(object theDocument, int hDC, double ModelToDC, int Rect)
        {
            eventLogTextBox.AppendText("OnAfterDocumentPrint");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnAfterDocumentSave(object theDocument)
        {
            eventLogTextBox.AppendText("OnAfterDocumentSave");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnAfterEnvironmentActivate(object theEnvironment)
        {
            eventLogTextBox.AppendText("OnAfterEnvironmentActivate");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnAfterNewDocumentOpen(object theDocument)
        {
            eventLogTextBox.AppendText("OnAfterNewDocumentOpen");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnAfterNewWindow(object theWindow)
        {
            eventLogTextBox.AppendText("OnAfterNewWindow");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnAfterWindowActivate(object theWindow)
        {
            eventLogTextBox.AppendText("OnAfterWindowActivate");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnBeforeCommandRun(int theCommandID)
        {
            eventLogTextBox.AppendText("OnBeforeCommandRun");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnBeforeDocumentClose(object theDocument)
        {
            eventLogTextBox.AppendText("OnBeforeDocumentClose");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnBeforeDocumentPrint(object theDocument, int hDC, double ModelToDC, int Rect)
        {
            eventLogTextBox.AppendText("OnBeforeDocumentPrint");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnBeforeDocumentSave(object theDocument)
        {
            eventLogTextBox.AppendText("OnBeforeDocumentSave");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnBeforeEnvironmentDeactivate(object theEnvironment)
        {
            eventLogTextBox.AppendText("OnBeforeEnvironmentDeactivate");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        public void OnBeforeQuit()
        {
            eventLogTextBox.AppendText("OnBeforeQuit");
            eventLogTextBox.AppendText(Environment.NewLine);

            _applicationEventWatcher.Dispose();
            _applicationEventWatcher = null;
            _application = null;
        }

        public void OnBeforeWindowDeactivate(object theWindow)
        {
            eventLogTextBox.AppendText("OnBeforeWindowDeactivate");
            eventLogTextBox.AppendText(Environment.NewLine);
        }

        #endregion

    }
}

File: 2014/AdvancedTrack/_03-Hands On - Handling Solid Edge Events/vb/Example1/Example1/Form1.vb
Imports SolidEdgeContrib
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace Example1
	Partial Public Class Form1
		Inherits Form

		Private _application As SolidEdgeFramework.Application = Nothing
		Private _applicationEventWatcher As ApplicationEventWatcher = Nothing
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			OleMessageFilter.Register()
		End Sub

		Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs) Handles MyBase.FormClosing
			If _applicationEventWatcher IsNot Nothing Then
				_applicationEventWatcher.Dispose()
				_applicationEventWatcher = Nothing
			End If
			_application = Nothing
		End Sub

		Private Sub exitToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs) Handles exitToolStripMenuItem.Click
			Close()
		End Sub

		Private Sub eventButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles eventButton.Click
			Try
				If eventButton.Checked Then
					If _application Is Nothing Then
						' On a system where Solid Edge is installed, the COM ProgID will be
						' defined in registry: HKEY_CLASSES_ROOT\SolidEdge.Application
						Dim t As Type = Type.GetTypeFromProgID(SolidEdge.PROGID.Application, throwOnError:= True)

						' Using the discovered Type, create and return a new instance of Solid Edge.
						_application = DirectCast(Activator.CreateInstance(type:= t), SolidEdgeFramework.Application)

						' Show Solid Edge.
						_application.Visible = True
					End If

					_applicationEventWatcher = New ApplicationEventWatcher(Me, _application)
				Else
					_applicationEventWatcher.Dispose()
					_applicationEventWatcher = Nothing
					_application = Nothing
				End If
			Catch ex As System.Exception
				MessageBox.Show(Me, ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
			End Try
		End Sub

		Private Sub clearButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles clearButton.Click
			eventLogTextBox.Clear()
		End Sub

		#Region "SolidEdgeFramework.ISEApplicationEvents"

		Public Sub OnAfterActiveDocumentChange(ByVal theDocument As Object)
			eventLogTextBox.AppendText("OnAfterActiveDocumentChange")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnAfterCommandRun(ByVal theCommandID As Integer)
			eventLogTextBox.AppendText("OnAfterCommandRun")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnAfterDocumentOpen(ByVal theDocument As Object)
			eventLogTextBox.AppendText("OnAfterDocumentOpen")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnAfterDocumentPrint(ByVal theDocument As Object, ByVal hDC As Integer, ByVal ModelToDC As Double, ByVal Rect As Integer)
			eventLogTextBox.AppendText("OnAfterDocumentPrint")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnAfterDocumentSave(ByVal theDocument As Object)
			eventLogTextBox.AppendText("OnAfterDocumentSave")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnAfterEnvironmentActivate(ByVal theEnvironment As Object)
			eventLogTextBox.AppendText("OnAfterEnvironmentActivate")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnAfterNewDocumentOpen(ByVal theDocument As Object)
			eventLogTextBox.AppendText("OnAfterNewDocumentOpen")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnAfterNewWindow(ByVal theWindow As Object)
			eventLogTextBox.AppendText("OnAfterNewWindow")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnAfterWindowActivate(ByVal theWindow As Object)
			eventLogTextBox.AppendText("OnAfterWindowActivate")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnBeforeCommandRun(ByVal theCommandID As Integer)
			eventLogTextBox.AppendText("OnBeforeCommandRun")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnBeforeDocumentClose(ByVal theDocument As Object)
			eventLogTextBox.AppendText("OnBeforeDocumentClose")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnBeforeDocumentPrint(ByVal theDocument As Object, ByVal hDC As Integer, ByVal ModelToDC As Double, ByVal Rect As Integer)
			eventLogTextBox.AppendText("OnBeforeDocumentPrint")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnBeforeDocumentSave(ByVal theDocument As Object)
			eventLogTextBox.AppendText("OnBeforeDocumentSave")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnBeforeEnvironmentDeactivate(ByVal theEnvironment As Object)
			eventLogTextBox.AppendText("OnBeforeEnvironmentDeactivate")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		Public Sub OnBeforeQuit()
			eventLogTextBox.AppendText("OnBeforeQuit")
			eventLogTextBox.AppendText(Environment.NewLine)

			_applicationEventWatcher.Dispose()
			_applicationEventWatcher = Nothing
			_application = Nothing
		End Sub

		Public Sub OnBeforeWindowDeactivate(ByVal theWindow As Object)
			eventLogTextBox.AppendText("OnBeforeWindowDeactivate")
			eventLogTextBox.AppendText(Environment.NewLine)
		End Sub

		#End Region

	End Class
End Namespace
